SPEC.md

```diff
@@ -5286,6 +5295,97 @@ Example output:
 </p>
 </details>
 
+### Runtime Access to Requirements, Hints, and Metadata
+
+The `requirements` and `hints` sections comprise resource requests to the execution engine. But these requests can be [specified or overridden at runtime](#specifying--overriding-requirements-and-hints), and the execution engine has some latitude in whether and how it fulfills them. Thus, the workflow developer may wish to know exactly what resources are available at runtime, such as:
+
+* What are the actual resource allocations. For example, a task may request at least `8 GB` of memory but may be able to use more memory if it is available.
```

Comment:
Nit - I think we should prefer `GiB` in most places in the spec as we often see people using `GB` and then wondering why they don't get enough memory.